src/ui/sign_message_nbgl.c

```diff
@@ -109,73 +67,83 @@ static nbgl_layoutTagValue_t *get_single_action_review_pair(uint8_t index) {
-    layout.pairs = NULL;  // to indicate that callback should be used
-    layout.callback = get_single_action_review_pair;
-    layout.startIndex = 0;
-    layout.nbPairs = transaction_steps_number;
-
-    nbgl_useCaseStaticReview(&layout,
-                             &review_final_long_press,
-                             option_reject_text,
-                             review_final_callback);
+static void review_choice(bool confirm) {
+    // Answer, display a status page and go back to main
+    // validate_transaction(confirm);
+    nbgl_reviewStatusType_t status_type;
+    if (confirm) {
+        sendResponse(set_result_sign_message(), ApduReplySuccess, false);
+        if (operation_type == TYPE_MESSAGE) {
+            status_type = STATUS_TYPE_MESSAGE_SIGNED;
+        } else {
+            status_type = STATUS_TYPE_TRANSACTION_SIGNED;
+        }
+        nbgl_useCaseReviewStatus(status_type, ui_idle);
+    } else {
+        sendResponse(0, ApduReplyUserRefusal, false);
+        if (operation_type == TYPE_MESSAGE) {
+            status_type = STATUS_TYPE_MESSAGE_REJECTED;
+        } else {
+            status_type = STATUS_TYPE_TRANSACTION_REJECTED;
+        }
+        nbgl_useCaseReviewStatus(status_type, ui_idle);
+    }
 }
 
 void start_sign_tx_ui(size_t num_summary_steps) {
-    review_final_long_press.text = "Sign transaction on\nSolana network?";
-    review_final_long_press.icon = &C_icon_solana_64x64;
-    review_final_long_press.longPressText = "Hold to sign";
-    review_final_long_press.longPressToken = 0;
-    review_final_long_press.tuneId = TUNE_TAP_CASUAL;
-
-    reject_choice_text = reject_choice_transaction;
-    go_back_text = go_back_transaction;
-    ask_reject_text = ask_reject_transaction;
-    success_text = success_transaction;
-    option_reject_text = option_reject_transaction;
+    // Set the transaction type
+    operation_type = TYPE_TRANSACTION;
 
     // Save steps number for later
     transaction_steps_number = num_summary_steps;
     last_step_is_ascii = false;
-    // start display
-    nbgl_useCaseReviewStart(&C_icon_solana_64x64,
-                            "Review transaction",
-                            NULL,
-                            "Reject transaction",
-                            start_review,
-                            rejectUseCaseChoice);
+
+    // Initialize the content structure
+    content.nbMaxLinesForValue = 0;
+    content.smallCaseForValue = false;
+    content.wrapping = true;
+    content.pairs = NULL;  // to indicate that callback should be used
+    content.callback = get_single_action_review_pair;
+    content.startIndex = 0;
+    content.nbPairs = transaction_steps_number;
+
+    // Start review
+    nbgl_useCaseReview(operation_type,
+                       &content,
+                       &C_icon_solana_64x64,
+                       "Review transaction",
+                       NULL,
+                       "Sign transaction on\nSolana network?",
```

Comment:
Do we really need the `\n`?
same question for `start_sign_offchain_message_ui` below.